src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -136,6 +136,33 @@ internal unsafe struct ManagedObjectWrapper
 
             internal CreateComInterfaceFlagsEx Flags;
 
+            public bool IsRooted
+            {
+                get
+                {
+                    ulong refCount = Interlocked.Read(ref RefCount);
+                    bool rooted = GetComCount(refCount) > 0;
+                    if (!rooted)
+                    {
+                        // TODO: global pegging state
+                        rooted = GetTrackerCount(refCount) > 0 && (Flags & CreateComInterfaceFlagsEx.IsPegged) != 0;
+                    }
+                    return rooted;
+                }
+            }
+
+            public ManagedObjectWrapperHolder? Holder
+            {
+                get
+                {
+                    IntPtr handle = HolderHandle;
```

Comment:
All other accesses to `HolderHandle` seem to be using interlocked APIs (which mean they also imply having volatile semantics for reads/writes). Should this read be volatile (or, just the field itself be marked as such) for consistency?